bundles/automation/org.eclipse.smarthome.automation.module.core/src/main/java/org/eclipse/smarthome/automation/module/core/handler/GenericEventTriggerHandler.java

```diff
@@ -93,7 +93,7 @@ public void receive(Event event) {
         if (callback != null) {
             logger.trace("Received Event: Source: " + event.getSource() + " Topic: " + event.getTopic() + " Type: "
                     + event.getType() + " Payload: " + event.getPayload());
-
+            // Check the topic for the source as the event source is not set for the most events.
```

Comment:
This is pretty misleading - the event.source is something completely different than the "source" configuration property in this handler (which means "object that this event refers to", but not "sender of the event").